core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -163,6 +166,9 @@ ManifestGroup planWith(ExecutorService newExecutorService) {
     DeleteFileIndex deleteFiles = deleteIndexBuilder.build();
 
     boolean dropStats = ManifestReader.dropStats(dataFilter, columns);
+    if (!deleteFiles.isEmpty()) {
+      select(Streams.concat(columns.stream(), ManifestReader.STATS_COLUMNS.stream()).collect(Collectors.toList()));
```

Comment:
This was needed to ensure the stats columns are projected for data files when there are delete files, even if the stats columns were not requested by the caller.